lib/resources/kernel_module.rb

```diff
@@ -36,19 +68,51 @@ def loaded?
       !found.nil?
     end
 
-    def version
-      if inspec.os.redhat? || inspec.os.name == 'fedora'
-        modinfo_cmd = "/sbin/modinfo -F version #{@module}"
-      else
-        modinfo_cmd = "modinfo -F version #{@module}"
-      end
+    def disabled?
+      !modprobe_output.match(%r{^install\s+#{@module}\s+/(s?)bin/(true|false)}).nil?
+    end
 
-      cmd = inspec.command(modinfo_cmd)
+    def disabled_via_bin_true?
+      !modprobe_output.match(%r{^install\s+#{@module}\s+/(s?)bin/true}).nil?
+    end
+
+    def disabled_via_bin_false?
+      !modprobe_output.match(%r{^install\s+#{@module}\s+/(s?)bin/false}).nil?
+    end
+
+    def blacklisted?
```

Comment:
Yes I was thinking that as well at first then I started to overthink it and did the above approach, but your right ( as was I ) an 'or' clause produces the same effect - i.e. return true if any of these return true else return false. ( :smack head ) 

Pushed an update to use the cleaner approach. 

Everything should be good to go.